src/prefect/blocks/core.py

```diff
@@ -790,6 +800,33 @@ async def _get_block_document(
 
         return block_document, block_document_name
 
+    @classmethod
+    @sync_compatible
+    @inject_client
+    async def _get_block_document_by_id(
+        cls,
+        block_document_id: str | uuid.UUID,
```

Comment:
we are still supporting Python 3.9 and so can't support this syntax quite yet! You'll need to use `typing.Union` here

Reply:
Thank you, fixed